FordLabs/PeopleMover
# Prod Push 7/24

## Issue
Connects #109 #116 #187 #197 #199 #200 #209

## What was done
- [x] #109 added back-end endpoint to get front-end environment variables
- [x] #116 #209 displays notes when hovering over person
- [x] #187 cleans out dates in calendar to only highlight dates with actual reassignments
- [x] #197 updates assignment effective dates to match when product start date is pushed to future
- [x] #199 fixes duplicate selections in filter dropdown
- [x] #200 removes default 'my product' when user creates a new space
- [x] #165 adds some prerequisites for adfs integration
- [x] minor linter updates

## How to test
#109
1. hitting '/api/config' endpoint should return list of accurate react environment variables.
2. no functional or visual changes to the user.

#116 #209 
1. add notes to a person.
2. the same person should have a notes icon next to their name in the assignment card. hovering over the icon should display the notes from step 1.
3. for people without notes, there should be no icon and no hovering feature.
4. for people who are unassigned (with or without notes), there should be no icon and no hovering feature.

#187
1. on a date where there are no reassignments, assign a person to a product. this reassignment should be in the reassigned drawer, and the date should be highlighted in the calendar.
2. repeat step 1 on a previous date.
3. on the date from step 2, the reassigned drawer should show this person's change, and the date should be highlighted in the calendar.
4. on the date from step 1, the reassigned drawer no longer shows this change, and the date is not highlighted in the calendar.

#197 
1. create a product, and assign a person. the reassignment drawer should display the change.
2. edit the product to start at a later date in the future.
3. the reassignment drawer should no longer display the change from step 1.
4. go to the date from step 2. the reassignment drawer should now display the person's assignment change.

#199
1. add multiple location tags and product tags.
2. select any of the tags in the filter dropdown.
3. products should be filtered accurately, and the filter dropdown should only display a check next to the tag selected in step 2.
4. edit the name of the selected tag.
5. the updated tag name should automatically be reflected in the product card(s) and in the filter dropdown.
6. the product(s) filtered should match those from step 3.

#200
1. create a space.
2. 'my product' should not automatically be there.
